ee/server/service/devices.go

```diff
@@ -16,6 +18,43 @@ func (svc *Service) RequestEncryptionKeyRotation(ctx context.Context, hostID uin
 	return svc.ds.SetDiskEncryptionResetStatus(ctx, hostID, true)
 }
 
+type migrateMDMDevicePayload struct {
+	Timestamp time.Time `json:"timestamp"`
+	Host      struct {
+		ID             uint   `json:"id"`
+		UUID           string `json:"uuid"`
+		HardwareSerial string `json:"hardware_serial"`
+	} `json:"host"`
+}
+
+func (svc *Service) TriggerMigrateMDMDevice(ctx context.Context, host *fleet.Host) error {
+	ac, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return err
+	}
+	if !ac.MDM.EnabledAndConfigured {
+		return fleet.NewMDMNotConfiguredError()
+	}
+	if !ac.MDM.MacOSMigration.Enable {
+		return fleet.NewBadGatewayError("macos_migration", ctxerr.New(ctx, "not enabled"))
+	}
+	if ac.MDM.MacOSMigration.WebhookURL == "" {
+		return fleet.NewBadGatewayError("macos_migration", ctxerr.New(ctx, "no webhook URL configured"))
+	}
```

Comment:
curious: this strikes me as bad requests, why bad gateway?

> The HyperText Transfer Protocol (HTTP) 502 Bad Gateway server error response code indicates that the server, while acting as a gateway or proxy, received an invalid response from the upstream server.

Reply:
I was debating this too. I talked myself out of a client error because there wasn't anything the client in this case (the device) couldn't do to turn the bad request into a good one. Bad gateway didn't seem like a great fit either but I figured it would spark the conversation lol.